perllib/FixMyStreet/Cobrand/Bexley/Waste.pm

```diff
@@ -15,6 +15,102 @@ has 'whitespace' => (
     default => sub { Integrations::Whitespace->new(%{shift->feature('whitespace')}) },
 );
 
+sub waste_fetch_events {
+    my ( $self, $params ) = @_;
+
+    my $missed_collection_service_property_id = 68;
+    my $db = FixMyStreet::DB->schema->storage;
+
+    my $missed_collection_reports = $self->problems->search(
+        {   external_id => { like => 'Whitespace%' },
+            state => [ FixMyStreet::DB::Result::Problem->open_states() ],
+        },
+        { order_by => 'id' },
+    );
+
+    while ( my $report = $missed_collection_reports->next ) {
+        print 'Fetching data for report ' . $report->id . "\n" if $params->{verbose};
+
+        my $worksheet_id = $report->external_id =~ s/Whitespace-//r;
+        my $worksheet
+            = $self->whitespace->GetFullWorksheetDetails($worksheet_id);
+
+        # Get info for missed collection
+        my $missed_collection_properties;
+        for my $service_properties (
+            @{  $worksheet->{WSServiceProperties}{WorksheetServiceProperty}
+                    // []
+            }
+        ) {
+            next
+                unless $service_properties->{ServicePropertyID}
+                == $missed_collection_service_property_id;
+
+            $missed_collection_properties = $service_properties;
+        }
+
+        my $whitespace_state_string
+            = $missed_collection_properties
+            ? $missed_collection_properties->{ServicePropertyValue}
+            : '';
+
+        my $new_state
+            = missed_collection_state_mapping()->{$whitespace_state_string};
+        unless ($new_state) {
+            print "  No new state, skipping\n" if $params->{verbose};
+            next;
+        }
+
+        next
+            unless $self->waste_check_last_update( $params, $report,
+            $new_state );
+
+        # Don't update state unless it's an allowed state
+        if ( FixMyStreet::DB::Result::Problem->visible_states()
+            ->{ $new_state->{fms_state} } )
+        {
+            print
+                "  Updating report to state '$new_state->{fms_state}' ('$new_state->{text}')\n"
+                if $params->{verbose};
+
+            $db->txn_do( sub {
+                my $comment = $report->comments->new(
+                    {
+                        created       => \'NOW()',
+                        # TODO Any IDs for worksheet updates?
+                        external_id => $report->external_id,
+                        problem       => $report,
+                        problem_state => $new_state->{fms_state},
+                        send_state  => 'processed',
+                        text        => $new_state->{text},
+                        user          => $self->body->comment_user,
+                    },
+                );
+                $comment->insert;
+                $report->state( $new_state->{fms_state} );
+                $report->lastupdate( $comment->created );
+                $report->update;
+            } );
```

Comment:
I see the existing `FixMyStreet::Roles::CobrandEcho::waste_fetch_events` method uses `Open311::GetServiceRequestUpdates` to process the updates from Echo.

Looking at the code there I think that's mostly so that it can use response templates, which isn't needed here currently because the response text is hard-coded, so I think this approach is ok for now, but mentioning in case this comes up in the future.